MobilityData/mobility-feed-api
# feat: make API publicly accessible for displaying UI

**Summary:**
This PR implements the feature of anonymous sign-in utilizing [Firebase Anonymous Authentication](https://firebase.google.com/docs/auth/web/anonymous-auth). This feature enables users to engage with the application without the need for creating a personal account. 

**Expected Behavior:**
Upon accessing the page, users will automatically be signed in anonymously. This anonymous sign-in also activates whenever a user, previously signed in (using email or any identity providers), signs out.

**Testing Tips:**
1. Switch to the feature branch of this PR (`feat/356`).
2. Execute the application and use console logging to retrieve the anonymous user's access token. Add the following line to file `auth-saga.ts` line 203 to view the `accessToken` value in the logs.
  ```typescript
console.log((currentUser as User)?.accessToken);
``` 
3. Attempt to access the API with the obtained token to verify functionality and access permissions for anonymous users.

This approach ensures seamless testing and validation of the new anonymous sign-in feature.
Please make sure these boxes are checked before submitting your pull request - thanks!
